src/panels/UserPanelsPanel/UserPanelsPanel.tsx

```diff
@@ -0,0 +1,134 @@
+import { useEffect, useState } from 'react';
+import {
+  Button,
+  Container,
+  Divider,
+  Group,
+  Select,
+  Text,
+  TextInput,
+  Title
+} from '@mantine/core';
+
+import { useOpenSpaceApi } from '@/api/hooks';
+import { UserPageIcon } from '@/icons/icons';
+import { useAppDispatch, useAppSelector } from '@/redux/hooks';
+import { intializeUserPages } from '@/redux/userpages/userpagesSlice';
+import { useWindowManagerProvider } from '@/windowmanagement/WindowLayout/hooks';
+
+import { UserPage } from './UserPanel';
+
+export function UserPanelsPanel() {
+  const { addWindow } = useWindowManagerProvider();
+
+  const [selectedPanel, setSelectedPanel] = useState<string | null>(null);
+  const [panelURL, setPanelURL] = useState<string>('');
+  const [urlPanelTitle, setUrlPanelTitle] = useState<string>('');
+
+  const luaApi = useOpenSpaceApi();
+  const isDataInitialized = useAppSelector((state) => state.userPages.isInitialized);
+
+  const localPanels = useAppSelector((state) => state.userPages.panels || []);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    // Collect all folder paths in the USER folder
+    const getUserPanels = async () => {
+      if (!luaApi) return;
+
+      // eslint-disable-next-line no-template-curly-in-string
+      const root = await luaApi.absPath('${USER}/webpanels');
+      const folders = await luaApi.walkDirectoryFolders(root);
+
+      if (!folders) return;
+
+      const slash = navigator.platform.indexOf('Win') > -1 ? '\\' : '/';
```

Comment:
My VS Code warns that `platform` is deprecated. 
![image](https://github.com/user-attachments/assets/0e77a8bb-970e-490b-b682-9c69008862f1)

So this should probably be implemented some other way.

Reply:
Use the path module and `path.dirname` instead?

https://stackoverflow.com/questions/818576/get-directory-of-a-file-name-in-javascript